tests/unit/test_simulated_stack.py

```diff
@@ -1147,8 +1147,44 @@ def testGetBioSample(self):
         for badId in self.getBadIds():
             self.verifyGetMethodFails(path, badId)
 
-    # TODO def testSearchPhenotypes(self):
+    def testSearchPhenotypeAssociationSets(self):
+        path = "/phenotypeassociationsets/search"
+        for dataset in self.dataRepo.getDatasets():
+            repoPaSetIds = []
+            for repoPaSet in dataset.getPhenotypeAssociationSets():
+                repoPaSetIds.append(repoPaSet.getId())
 
-    # TODO def testSearchPhenotypeAssociationSets(self):
+            request = protocol.SearchPhenotypeAssociationSetsRequest()
+            request.dataset_id = dataset.getId()
+            responseData = self.sendSearchRequest(
+                path, request,
+                protocol.SearchPhenotypeAssociationSetsResponse)
+            for clientPaSet in responseData.phenotype_association_sets:
+                self.assertTrue(clientPaSet.id in repoPaSetIds)
 
-    # TODO def testSearchGenotypePhenotypes(self):
+    def testSearchPhenotypes(self):
+        path = "/phenotypes/search"
+        for dataset in self.dataRepo.getDatasets():
+            for repoPaSet in dataset.getPhenotypeAssociationSets():
+                for repoAssoc in repoPaSet.getAssociations():
+                    request = protocol.SearchPhenotypesRequest()
+                    request.phenotype_association_set_id = repoPaSet.getId()
+                    request.id = repoAssoc.phenotype.id
+                    responseData = self.sendSearchRequest(
+                        path, request,
+                        protocol.SearchPhenotypesResponse)
+                    for clientPhenotype in responseData.phenotypes:
+                        self.assertEqual(clientPhenotype, repoAssoc.phenotype)
+
+    def testSearchGenotypePhenotypes(self):
+        path = "/genotypephenotypes/search"
+        for dataset in self.dataRepo.getDatasets():
+            for repoPaSet in dataset.getPhenotypeAssociationSets():
```

Comment:
You can replace this double loop with a single loop by using `allPhenotypeAssociationSets`